### PR TITLE
Fixed some issues (see details below) with TYPO3 7 LTS (7.6.9)

### PR DESCRIPTION
**Pull request by chuenniger**
See commits below. Some fixes for the TYPO3 7 LTS compatibility (7.6.9)

Fixed: Scheduler message ($className "\TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer" must not start with a backslash.'")
Fixed: BE Warnings with empty flexform entries
* Core: Error handler (BE): PHP Warning: Illegal string offset 'config' in /mnt/data/www/lib/typo3/typo3_src-7.6.9/typo3/sysext/backend/Classes/Form/FormDataProvider/TcaInputPlaceholders.php line 45
* Core: Error handler (BE): PHP Warning: Illegal string offset 'default' in /mnt/data/www/lib/typo3/typo3_src-7.6.9/typo3/sysext/backend/Classes/Form/FormDataProvider/DatabaseRowDefaultValues.php line 59 Added: .gitignore

### CODE REVIEW

File: res/js/tx.caretaker.NodeTree.js [hunk 1]
```diff
@@ -354,7 +355,7 @@ tx.caretaker.NodeTree = Ext.extend(Ext.tree.TreePanel, {
 				"tx_caretaker[isInInstance]": (node.attributes.type == 'instance' ? 1 : 0),
 				"tx_caretaker[type]": 'test',
 				"tx_caretaker[storagePid]": this.storagePid,
-				"tx_caretaker[returnUrl]": encodeURIComponent(parent.list_frame.document.location)
```
Comment (etobi) [inline res/js/tx.caretaker.NodeTree.js]: Why did you remove these encodeURIComponent()s?
I'm not sure if they are necessary, just wonder why you removed it

File: Classes/services/notifications/advanced/exitpoints/class.tx_caretaker_NotificationBaseExitPoint.php [hunk 2]
```diff
@@ -83,7 +83,7 @@ public function init(array $config) {
 		}
 
 		$this->cObj = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(
-			'\\TYPO3\\CMS\\Frontend\\ContentObject\\ContentObjectRenderer'
+			'TYPO3\\CMS\\Frontend\\ContentObject\\ContentObjectRenderer'
```
Comment (etobi) [inline Classes/services/notifications/advanced/exitpoints/class.tx_caretaker_NotificationBaseExitPoint.php]: 👍

File: Classes/ajax/class.tx_caretaker_NodeInfo.php [hunk 3]
```diff
@@ -248,7 +250,12 @@ public function ajaxGetNodeGraph($params, &$ajaxObj) {
 		if ($node_id && $node) {
 			$result_range = $node->getTestResultRange($date_start, $date_stop);
 			if ($result_range->count()) {
-				$filename = 'typo3temp/caretaker/charts/' . $node_id . '_' . $duration . '.png';
+				if (!is_dir(PATH_site . self::PATH_CHARTS)) {
+					if (!mkdir(PATH_site . self::PATH_CHARTS, 0770, true)) {
+						throw new \TYPO3\CMS\Core\Cache\Exception('can\'t create path "'.PATH_site.self::PATH_CHARTS.'"', 1465993775);
+					}
+				}
+				$filename = self::PATH_CHARTS . DIRECTORY_SEPARATOR . $node_id . '_' . $duration . '.png';
```
Comment (etobi) [inline Classes/ajax/class.tx_caretaker_NodeInfo.php]: 👍

File: Classes/ajax/class.tx_caretaker_Utility.php [hunk 4]
```diff
@@ -118,7 +118,10 @@ public function getModuleUrl() {
 				} else {
 					$moduleUrl = BackendUtility::getModuleUrl(
 						'record_edit',
-						array('edit' => array($table => array($node => 'edit')))
+						array(
+							'edit' => array($table => array($node => 'edit')),
+							'returnUrl' => $requestParameters['returnUrl']
```
Comment (etobi) [inline Classes/ajax/class.tx_caretaker_Utility.php]: 👍

File: res/js/tx.caretaker.NodeTree.js (hunk 1 above)
Comment (chuenniger) [inline res/js/tx.caretaker.NodeTree.js]: Hi,

because the url was encoded two times in backend an than it was not possible to go back or something else.

Issue describtion
* Go to caretaker view in BE
* Edit a test case
* click "go back" (not save or something else) 
* This is not possible at the moment

Comment (etobi) [inline res/js/tx.caretaker.NodeTree.js]: 👍

**Comment by etobi** [comment]
Thanks for your PR. I will release a bugfix version within the next days.
